src/NServiceBus.Transport.RabbitMQ/Routing/DirectRoutingTopology.cs

```diff
@@ -8,40 +10,93 @@
     /// <summary>
     /// Route using a static routing convention for routing messages from publishers to subscribers using routing keys.
     /// </summary>
-    class DirectRoutingTopology : IRoutingTopology
+    public class DirectRoutingTopology : IRoutingTopology
```

Comment:
The only alternative would be to use enums but this makes is awkward when combined with a custom topology because that would have to be an instance.

In alternative could be some static field e.g. have `Topologies` static class with fields for direct and conventional topology. Then the user could do

```
config.RoutingTopology = Topologies.Direct;
config.RoutingTopology = Topologies.Conventional;
config.RoutingTopology = new MyCustomTopology();
config.RoutingTopology = new ConventionalRoutingTopology(false, oneConvention, anotherConvention)
```

but since the most common case of conventional topology is covered by default settings, I think that would be excessive complexity. If a user knows what direct topology is, they can surely be bothered by instantiating a public class.

Reply:
Hard disagree here. If we are "improving" the transport seam, why is leading to APIs that are worse than what we had before?

Reply:
I disagree that this is worse if you take into account all the usage scenarios. The advantage of the new API is that is allows users to inspect the state of the configuration object.